clients/src/main/java/org/apache/kafka/clients/consumer/internals/SubscriptionState.java

```diff
@@ -76,6 +76,9 @@
     /* the partitions that are currently assigned, note that the order of partition matters (see FetchBuilder for more details) */
     private final PartitionStates<TopicPartitionState> assignment;
 
+    /* the number of partitions that are currently assigned available in a thread safe manner */
+    private volatile int assignmentSize;
```

Comment:
I think I'd do this by wrapping `PartitionStates` in a static inner class. That would keep the logic in one place. But it would be good to get @hachikuji's thoughts as well.

Reply:
I agree that seems a little cleaner.